File: 002/euler002.py
#!/usr/bin/python
# 2009.04.02

# Find the sum of all the even-valued terms in the Fibonacci sequence
# which do not exceed four million.

fib = [0, 1];
sum = 0;

# Create list of fibonacci numbers <= 4 million
for i in range(2,1000): # How do you know 1000 is enough?
    if fib[-1] <= 4000000:
        fib.append(fib[i-1] + fib[i-2]);

#print "\nfib\n", fib;
#print "\nfib[2:-2]\n", fib[2:-1];

for i in fib[2:-1]: # Skips 0, 1, and the last (which is > 4 million)
    if i % 2 == 0:  # Excludes odds
        sum += i;   # Adds evens

print sum;

File: 002/euler002.c
/* Steve Phillips
   2009.07.22 (Really started and finished 2009.07.24)

   Find the sum of all the even-valued terms in the sequence which do
   not exceed four million.
*/

#include <stdio.h>
#define BIGNUM 4000000

int a, b, c;
int evensum = 2;

int main(int argc, char **argv)
{
  printf("The sum of even fibonacci numbers less than %d is %d\n",
	 BIGNUM, sumOfEvenFib(BIGNUM));

  return 0;
}

// Returns the sum of all even fibonacci numbers <= BIGNUM
int sumOfEvenFib(int num) {
  num = BIGNUM;
  a = 1;
  b = 2;
  c = b + a;

  while (c <= BIGNUM) {
    //    printf("c: %d\n", c);
    if (c % 2 == 0)
      evensum += c;
    a = b;
    b = c;
    c = b + a;
  }
  return evensum;
}
